core/corehttp/gateway_handler.go

```diff
@@ -378,7 +379,9 @@ func (i *gatewayHandler) serveFile(w http.ResponseWriter, req *http.Request, nam
 		if ctype == "" {
 			buf := make([]byte, 512)
 			n, _ := io.ReadFull(content, buf[:])
-			ctype = http.DetectContentType(buf[:n])
+			// uses https://github.com/gabriel-vasile/mimetype library to determine the content type.
+			// Fixes https://github.com/ipfs/go-ipfs/issues/7252
+			ctype = mimetype.Detect(buf[:n]).String()
```

Comment:
I think we could actually use `DetectReader(content)` directly. This will read at most 3072 bytes. It is more than the current 512 but worth it if that gives you an accurate type (and also means it will read less for things that can be detected with less, which are hopefully most).

Reply:
okay. I assume we don't have to seek to the start if we're using `DetectReader(content)`?

Reply:
@hsanjuan - fixed via 9201b1d.